server/src/fstar_handlers.ts

```diff
@@ -298,9 +140,13 @@ function handleIdeProgress(textDocument: TextDocument, contents: IdeProgress, la
 	}
 }
 
-// If we get errors and warnings from F*, we send them to VSCode
-// as diagnostics, which will show them as squiggles in the editor
-function handleIdeDiagnostics(textDocument: TextDocument, response: IdeError[], lax: boolean, server: Server) {
+// If we get errors and warnings from F*, we send them to VSCode as diagnostics,
+// which will show them as squiggles in the editor
+//
+// TODO(klinvill): with my latest edits, this function is only called for
+// fullBufferQuery requests. Should it also be called for other requests? What
+// about ones for which responses aren't expected?
```

Comment:
Only full buffer requests produce messages that should be shown as diagnostics, so yes.